homeassistant/components/rainbird/quality_scale.yaml

```diff
@@ -0,0 +1,85 @@
+rules:
+  # Bronze
+  config-flow:
+    status: todo
+    comment: Some fields are missing data descriptions.
+  brands: done
+  dependency-transparency: done
+  common-modules: done
```

Comment:
This integration would benefit from using a shared entity though

Reply:
Would it? I'm yet to be convinced of the value of doing this as I tend to prefer composition over inheritance and find it difficult to read code with extra layers of inheritance with little value add.

My impression is that only thing shared is `_attr_has_entity_name = True`. Some entieies have the same code for `self._attr_device_info = coordinator.device_info` but not all of them given there a a couple kinds of devices. The unique id logic has the same prefix, but may different depending on the platform details like sensor key.

My main worry is that additional inheritance without added value makes integration code difficult to read when reading an individual entities code. However, I can definitely see it helps you for a review like this when you are trying to answer a question like "do all entities in this platform set an entity name" without having to read each of them -- however, I think programmatic enforcement or framework support could help.

Reply:
Oh I got the idea that it was always using the same device. The creation of the DeviceInfo can then be moved to the shared entity, because I do think the device is an entity concern and not a coordinator.

I don't really think enforcement is necessary, it also allows for flexibility.